diff -urw xc-orig/programs/Xserver/hw/xwin/winclipboard.h xc/programs/Xserver/hw/xwin/winclipboard.h
--- xc-orig/programs/Xserver/hw/xwin/winclipboard.h	2009-03-18 18:13:40.000000000 +0100
+++ xc/programs/Xserver/hw/xwin/winclipboard.h	2011-12-11 22:25:09.726239465 +0100
@@ -72,8 +72,8 @@
 
 
 /* Clipboard module constants */
-#define WIN_CLIPBOARD_WINDOW_CLASS		"xwinclip"
-#define WIN_CLIPBOARD_WINDOW_TITLE		"xwinclip"
+#define WIN_CLIPBOARD_WINDOW_CLASS		"OpenNXWinClip"
+#define WIN_CLIPBOARD_WINDOW_TITLE		"OpenNXWinClip"
 #ifdef HAS_DEVWINDOWS
 # define WIN_MSG_QUEUE_FNAME			"/dev/windows"
 #endif
diff -urw xc-orig/programs/Xserver/hw/xwin/winclipboardwndproc.c xc/programs/Xserver/hw/xwin/winclipboardwndproc.c
--- xc-orig/programs/Xserver/hw/xwin/winclipboardwndproc.c	2009-03-18 18:13:40.000000000 +0100
+++ xc/programs/Xserver/hw/xwin/winclipboardwndproc.c	2011-12-11 22:25:09.727239465 +0100
@@ -60,6 +60,8 @@
 /* BPS - g_hwndClipboard needed for X app->Windows paste fix */
 extern HWND		g_hwndClipboard;
 
+extern int      g_iClipFilter;
+
 /* 
  * Local function prototypes
  */
@@ -155,6 +157,7 @@
  * Windows app. TODO - Perhaps move to win.h with the other WM_USER messages.
  */
 #define WM_USER_PASTE_COMPLETE		(WM_USER + 1003)
+#define WM_USER_SET_FILTER  		(WM_USER + 1004)
 
 LRESULT CALLBACK
 winClipboardWindowProc (HWND hwnd, UINT message, 
@@ -166,6 +169,12 @@
   /* Branch on message type */
   switch (message)
     {
+    case WM_USER_SET_FILTER:
+        if ((0 < wParam) && (4 > wParam)) {
+            g_iClipFilter = (int)wParam;
+            ErrorF ("Setting ClipFilter to %d\n", g_iClipFilter);
+        } 
+        return 0;
     case WM_DESTROY:
       {
 	winDebug ("winClipboardWindowProc - WM_DESTROY\n");
diff -urw xc-orig/programs/Xserver/hw/xwin/winclipboardwrappers.c xc/programs/Xserver/hw/xwin/winclipboardwrappers.c
--- xc-orig/programs/Xserver/hw/xwin/winclipboardwrappers.c	2011-12-11 02:06:30.189499447 +0100
+++ xc/programs/Xserver/hw/xwin/winclipboardwrappers.c	2011-12-11 22:25:09.728239464 +0100
@@ -74,6 +74,7 @@
 extern winDispatchProcPtr	winProcQueryTreeOrig;
 extern winDispatchProcPtr	winProcSetSelectionOwnerOrig;
 
+extern int g_iClipFilter;
 
 /*
  * Wrapper for internal QueryTree function.
@@ -530,7 +531,21 @@
   SetClipboardData (CF_TEXT, NULL);
 
   /* Save handle to last owned selection */
+  switch (g_iClipFilter) {
+    case CLIPFILTER_PRIMARY:
+      if (XA_PRIMARY == stuff->selection) {
   g_atomLastOwnedSelection = stuff->selection;
+      }
+      break;
+    case CLIPFILTER_CLIPBOARD:
+      if (MakeAtom ("CLIPBOARD", 9, TRUE) == stuff->selection) {
+        g_atomLastOwnedSelection = stuff->selection;
+      }
+      break;
+    case CLIPFILTER_BOTH:
+      g_atomLastOwnedSelection = stuff->selection;
+      break;
+  }
 
   /* Release the clipboard */
   if (!CloseClipboard ())
diff -urw xc-orig/programs/Xserver/hw/xwin/winclipboardxevents.c xc/programs/Xserver/hw/xwin/winclipboardxevents.c
--- xc-orig/programs/Xserver/hw/xwin/winclipboardxevents.c	2009-03-18 18:13:40.000000000 +0100
+++ xc/programs/Xserver/hw/xwin/winclipboardxevents.c	2011-12-11 22:25:09.730239464 +0100
@@ -33,6 +33,8 @@
 #endif
 #include "winclipboard.h"
 
+#define CLPDEBUG 0
+
 
 /*
  * References to external symbols
@@ -100,7 +102,7 @@
 	   */
 
 	case SelectionRequest:
-#if 0
+#if CLPDEBUG
 	  {
 	    char			*pszAtomName = NULL;
 	    
@@ -445,7 +447,7 @@
 	   */ 
 
 	case SelectionNotify:
-#if 0
+#if 1
 	  ErrorF ("winClipboardFlushXEvents - SelectionNotify\n");
 	  {
 	    char		*pszAtomName;
@@ -468,7 +470,7 @@
 	    {
 	      if (event.xselection.target == XA_STRING)
 		{
-#if 0
+#if CLPDEBUG
 		  ErrorF ("winClipboardFlushXEvents - SelectionNotify - "
 			  "XA_STRING\n");
 #endif
@@ -476,7 +478,7 @@
 		}
 	      else if (event.xselection.target == atomUTF8String)
 		{
-#if 0
+#if CLPDEBUG
 		  ErrorF ("winClipboardFlushXEvents - SelectionNotify - "
 			  "Requesting conversion of UTF8 target.\n");
 #endif
@@ -502,7 +504,7 @@
 #ifdef X_HAVE_UTF8_STRING
 	      else if (event.xselection.target == atomCompoundText)
 		{
-#if 0
+#if CLPDEBUG
 		  ErrorF ("winClipboardFlushXEvents - SelectionNotify - "
 			  "Requesting conversion of CompoundText target.\n");
 #endif
@@ -556,7 +558,7 @@
 	      break;
 	    }
 
-#if 0
+#if CLPDEBUG
 	  ErrorF ("SelectionNotify - returned data %d left %d\n",
 		  xtpText.nitems, ulReturnBytesLeft);
 #endif
@@ -582,7 +584,7 @@
 	      break;
 	    }
 
-#if 0
+#if CLPDEBUG
 	    {
 	      char		*pszAtomName = NULL;
 
diff -urw xc-orig/programs/Xserver/hw/xwin/winglobals.c xc/programs/Xserver/hw/xwin/winglobals.c
--- xc-orig/programs/Xserver/hw/xwin/winglobals.c	2009-03-18 18:13:40.000000000 +0100
+++ xc/programs/Xserver/hw/xwin/winglobals.c	2011-12-11 22:25:09.731239464 +0100
@@ -115,6 +115,7 @@
 void			*g_pClipboardDisplay = NULL;
 Window			g_iClipboardWindow = None;
 Atom			g_atomLastOwnedSelection = None;
+int             g_iClipFilter = CLIPFILTER_BOTH;
 #endif
 
 
diff -urw xc-orig/programs/Xserver/hw/xwin/win.h xc/programs/Xserver/hw/xwin/win.h
--- xc-orig/programs/Xserver/hw/xwin/win.h	2011-12-11 02:06:30.192499447 +0100
+++ xc/programs/Xserver/hw/xwin/win.h	2011-12-11 22:25:09.733239464 +0100
@@ -808,6 +808,10 @@
 
 void
 winFixClipboardChain (void);
+
+# define CLIPFILTER_PRIMARY   1
+# define CLIPFILTER_CLIPBOARD 2
+# define CLIPFILTER_BOTH      3
 #endif
 
 
diff -urw xc-orig/programs/Xserver/hw/xwin/winprocarg.c xc/programs/Xserver/hw/xwin/winprocarg.c
--- xc-orig/programs/Xserver/hw/xwin/winprocarg.c	2011-12-11 02:06:30.194499447 +0100
+++ xc/programs/Xserver/hw/xwin/winprocarg.c	2011-12-12 12:29:43.119998361 +0100
@@ -50,6 +50,7 @@
 #ifdef XWIN_CLIPBOARD
 extern Bool			g_fUnicodeClipboard;
 extern Bool			g_fClipboard;
+extern int          g_iClipFilter;
 #endif
 extern int			g_iLogVerbose;
 extern char *			g_pszLogFile;
@@ -917,9 +918,27 @@
   if (IS_OPTION ("-clipboard"))
     {
       g_fClipboard = TRUE;
+      int iArgsProcessed = 1;
 
+      /* Grab the optional filter value */
+      if (i + 1 < argc) {
+          iArgsProcessed++;
+	      sscanf (argv[i + 1], "%d", &g_iClipFilter);
+          if ((CLIPFILTER_PRIMARY > g_iClipFilter) || (CLIPFILTER_BOTH < g_iClipFilter)) {
+              g_iClipFilter = CLIPFILTER_BOTH;
+          }
+	      if (0 == strcmp(argv[i + 1], "primary")) {
+              g_iClipFilter = CLIPFILTER_PRIMARY;
+          }
+	      if (0 == strcmp(argv[i + 1], "clipboard")) {
+              g_iClipFilter = CLIPFILTER_CLIPBOARD;
+          }
+	      if (0 == strcmp(argv[i + 1], "both")) {
+              g_iClipFilter = CLIPFILTER_BOTH;
+          }
+      }
       /* Indicate that we have processed this argument */
-      return 1;
+      return iArgsProcessed;
     }
 #endif
 
